### ISSUE TITLE
Feedbin starred articles inaccurately being flagged as unread

### ISSUE THREAD

**Issue by philviso** [brentsimmons/NetNewsWire#868]
Starring an article in another app, usually goes like this for me...

1. In Reeder for iPhone, swipe an article in my timeline to star it for reading later
2. "Mark all as read" in my timeline

(Article was starred, but also marked as read afterward)

When I open NetNewsWire (usually much later than the last time I saved an article), after it does the startup refresh, it will show those articles as both starred and unread. This happens nearly every time I open the app.

Here are some screenshots

<img width="1405" alt="Screen Shot 2019-08-21 at 7 29 43 AM" src="https://user-images.githubusercontent.com/347423/63432040-e8130b00-c3e5-11e9-8494-4c6e854e0afc.png">

<img width="1405" alt="Screen Shot 2019-08-21 at 7 32 38 AM" src="https://user-images.githubusercontent.com/347423/63432045-e9dcce80-c3e5-11e9-9c96-9d108edf31ec.png">

At some point, it does another automatic refresh or UI update or something, and the problem corrects itself. For example, while I was typing this bug report, I just went back to NetNewsWire and noticed that the problem corrected itself.

<img width="1405" alt="Screen Shot 2019-08-21 at 7 38 02 AM" src="https://user-images.githubusercontent.com/347423/63432382-a9ca1b80-c3e6-11e9-9427-b6a4e4db545c.png">

**Comment by vincode-io** [comment]
I wonder if Reeder isn't sending your Mark All as Read status updates immediately.  Is it possible that you are closing your phone after doing the Mark All as Read without sending Reeder to the background?  If so, it might have to wait for a background refresh to get your Mark All as Read statuses to Feedbin.

I tried to recreate your problem using the following steps and wasn't able to.

In Reeder:
Step 1:  Swipe to mark an article as read
Step 2: Use the context menu to mark the article as read
Step 3: Go to the home screen to send Reeder into the background

In NetNewsWire:
Step 1: Launch NetNewsWire
Step 2: Watch the refresh process run
Step 3: Verify that the article is both now starred and unread

If you have some series of steps that I could perform to replicate the problem, it would really help me.

**Comment by philviso** [comment]
It's possible that it's related to Reeder behavior. I'll try playing around with some variations in my behavior and see if it behaves any differently

A few times when this has happened in NNW, I've tried opening other RSS readers on my Mac (ReadKit and Reeder), and both of them are properly showing the starred articles as being read.

**Comment by vincode-io** [comment]
I just saw this happen where I had unread items pop up in my All Unread smart feed.

What I think is happening is that we are getting the articles in from Feedbin, persisting them, and then sending out a Notification that they arrived.  This triggers the All Unread smart feed to reload at roughly the same time as we are doing the fetch to get the article statuses.  The All Unread smart feed reloads then get the Notification for the updated statuses.  That would explain how you got unread and starred items come into  your All Unread smart feed.

I think that if we grab the statuses before grabbing the articles in the sync process that this problem will go away.

**Comment by brentsimmons** [comment]
That makes sense — ideally we have the statuses in place before we get the articles.

**Comment by vincode-io** [comment]
I was partially wrong about what was going on here.  Changing to get the statuses first did make the All Unread smartfeed work much better.  It also made @philviso's issue 100% reproducible.

When we were getting starred statuses before we got the article, we were creating the status row in the database with the read column as false.  It needs to be true for Feedbin, since the status they send us are for unread statuses.  We have to default read to being true.

I ended up killing two birds with one stone on this one.
